### DISCUSSION TITLE
Guidance on Adapting PyBaMM's get_min_max_stoichiometries

### DISCUSSION THREAD

**Discussion by kimo707** [pybamm-team/PyBaMM#3956] [Q&A] [answered]
Hello, I'm currently developing code aimed at assessing the state of health (SOH) of a battery. In the process, I encountered an issue. Previously, I employed the ElectrodeSOH model from PyBaMM's lithium-ion battery module in the following manner:

esoh_model = pybamm.lithium_ion.ElectrodeSOH()
esoh_sim = pybamm.Simulation(esoh_model, parameter_values=parameter_values)
esoh_sol = esoh_sim.solve(
            [0],
            inputs={"V_min": Vmin, "V_max": Vmax, "C_n": Cn, "C_p": Cp, "n_Li": n_Li},

However, this methodology appears to be deprecated as of version 23.4.1 and is also not present in the latest version. As an alternative strategy, I began utilizing the get_min_max_stoichiometries(parameter_values, param) function to ascertain the minimum and maximum stoichiometries (x_0, x_100, y_0, y_100), but I am uncertain about how to effectively adapt this function. Specifically, I wish to update the input data with {"Q_n": Q_n, "Q_p": Q_p, "Q_Li": n_Li} before proceeding, yet I am unclear on the appropriate manner to integrate these variables into the get_min_max_stoichiometries function to refresh Q_n and Q_p prior to recalculating (x_0, x_100, y_0, y_100). Could you offer advice on how to modify this function, or suggest an alternative method to achieve this?

**Comment by valentinsulzer** [comment, marked as answer]
Check https://docs.pybamm.org/en/latest/source/examples/notebooks/models/electrode-state-of-health.html

"C_n" = "Q_n", "C_p" = "Q_p", "Q_Li" = F / 3600 * "n_Li"

**Reply by kimo707** [reply]
Thanks for your reply
